bindings/python/pymongoarrow/builders.pyi

```diff
@@ -20,28 +20,29 @@ cdef class _ArrayBuilderBase:
 
 
 cdef class Int32Builder(_ArrayBuilderBase):
+    type_marker = _BsonArrowTypes.int32
```

Comment:
What does `type_marker` do?

Reply:
It's used to replace the `type_map` lookup in process_bson_stream:
```python
- ftype = type_map[key]
+ ftype = builder.type_marker
```
This results in a sizeable speedup. After this change we can probably remove `type_map` altogether.

Reply:
Got it.